tests/GraphBLAS-sharp.Tests/BackendCommonTests/COOMatrixEwiseAddTests.fs

```diff
@@ -107,92 +88,146 @@ let testCases =
+                    let actualValues = Array.zeroCreate res.Values.Length
+
+                    let _ =
+                        q.Post(Msg.CreateToHostMsg(res.RowPointers, actualRows))
+
+                    let _ =
+                        q.Post(Msg.CreateToHostMsg(res.Columns, actualColumns))
+
+                    let _ =
+                        q.PostAndReply(fun ch -> Msg.CreateToHostMsg(res.Values, actualValues, ch))
+
+                    q.Post(Msg.CreateFreeMsg<_>(res.Columns))
+                    q.Post(Msg.CreateFreeMsg<_>(res.RowPointers))
+                    q.Post(Msg.CreateFreeMsg<_>(res.Values))
+
+                    { CSRMatrix.RowCount = res.RowCount
+                      ColumnCount = res.ColumnCount
+                      RowPointers = actualRows
+                      ColumnIndices = actualColumns
+                      Values = actualValues }
 
-    [ testProperty "Correctness test on random int arrays"
-      <| (let replicate = COOMatrix.eWiseAdd context <@ (+) @>
-          let getReplicateFun = getMAddFun replicate
+                logger.debug (
+                    eventX "Actual is {actual}"
+                    >> setField "actual" (sprintf "%A" actual)
+                )
 
-          fun (array: int [,]) i1 i2 ->
-              let m1, m2 = genCOOMatricesToAdd array i1 i2
-              makeTest getReplicateFun (+) 0 m1 m2)
+                checkResult op zero baseMtx1 baseMtx2 (MatrixCSR(actual))
 
-    (*testProperty "Correctness test on random bool arrays"
-      <| (let replicate = ClArray.replicate context
-          let getReplicateFun = getReplicateFun replicate
+        | _ -> failwith "No other types of matrices tested yet."
 
-          fun (array: array<bool>) -> makeTest getReplicateFun array)
+    [ testProperty "Correctness test on random int arrays"
+      <| (fun size -> makeTest context (PairOfSparseMatricesOfEqualSize.IntType()) size COO (+) <@ (+) @> 0)
 
-      testProperty "Correctness test on random float arrays"
-      <| (let replicate = ClArray.replicate context
-          let getReplicateFun = getReplicateFun replicate
+      testProperty "Correctness test on random bool arrays"
+      <| (fun size -> makeTest context (PairOfSparseMatricesOfEqualSize.BoolType()) size COO (||) <@ (||) @> false)
 
-          fun (array: array<float>) -> makeTest getReplicateFun array)
+      testProperty "Correctness test on random float arrays"
+      <| (fun size -> makeTest context (PairOfSparseMatricesOfEqualSize.FloatType()) size COO (+) <@ (+) @> 0.0)
 
       testProperty "Correctness test on random byte arrays"
-      <| (let replicate = ClArray.replicate context
-          let getReplicateFun = getReplicateFun replicate
-
-          fun (array: array<byte>) -> makeTest getReplicateFun array)*)
-
-     ]
+      <| (fun size -> makeTest context (PairOfSparseMatricesOfEqualSize.ByteType()) size COO (+) <@ (+) @> (byte 0)) ]
```

Comment:
```byte 0``` can be replaced with ```0uy``` (https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/literals)